src/main.ts

```diff
@@ -334,30 +382,17 @@ export default class NewZettel extends Plugin {
             : selectionPos.anchor.line < selectionPos.head.line; // else they are not on the same line and just check if anchor is before head
 
         // if anchorCorrect use as is, else switch
-        const virtualAnchor = anchorCorrect
-          ? selectionPos.anchor
-          : selectionPos.head;
-        const virtualHead = anchorCorrect
-          ? selectionPos.head
-          : selectionPos.anchor;
-        editor!.replaceRange(
-          " ".repeat(spaceBefore) + newLink(title) + " ".repeat(spaceAfter),
-          virtualAnchor,
-          virtualHead
-        );
-        this.makeNote(nextPath(title), title, fileLink, true, openNewFile);
+        const virtualAnchor = anchorCorrect ? selectionPos.anchor : selectionPos.head;
+        const virtualHead = anchorCorrect ? selectionPos.head : selectionPos.anchor;
+        // editor!.replaceRange(" ".repeat(spaceBefore) + newLink(title) + " ".repeat(spaceAfter), virtualAnchor, virtualHead);
+        this.makeNote(nextPath(title), title, fileLink, true, openNewFile, () => {
+          editor!.replaceRange(" ".repeat(spaceBefore) + newLink(title) + " ".repeat(spaceAfter), virtualAnchor, virtualHead);
```

Comment:
This seems less readable compressed...

Reply:
Mmhh... don't know how to make it more readable, since I just copied the original replace method into a callback method. Could declare the callback method beforehand, but I don't think that makes stuff more readable necessarily.